packages/features/bookings/Booker/utils/use-brand-colors.tsx

```diff
@@ -0,0 +1,26 @@
+import { useEffect } from "react";
+
+import useGetBrandingColours from "@calcom/lib/getBrandColours";
+import useTheme from "@calcom/lib/hooks/useTheme";
+import { useCalcomTheme } from "@calcom/ui";
+
+export const useBrandColors = ({
+  brandColor,
+  darkBrandColor,
+  theme,
+}: {
+  brandColor?: string;
+  darkBrandColor?: string;
+  theme?: string | null;
+}) => {
+  const brandTheme = useGetBrandingColours({
+    lightVal: brandColor,
+    darkVal: darkBrandColor,
+  });
+  useCalcomTheme(brandTheme);
+  const { setTheme } = useTheme(theme);
+
+  useEffect(() => {
+    if (theme) setTheme(theme);
+  }, [setTheme, theme]);
+};
```

Comment:
Yeah great question. I took the logic from the current booker, and actually not really sure if it's used anywhere else (and the current booker will be gone soon). 

@sean-brydon  what do you think about this? It pretty much combines most of your hooks into one hook – so perhaps it indeed makes sense to reuse it. On top of that I also added an effect that triggers the `setTheme` again, because in our case the `theme` is undefined during load, which would cause to set the default and not update it anymore later.